### DISCUSSION TITLE
CLI Design of camelCase Options.

### DISCUSSION THREAD

**Discussion by 7086cmd** [rolldown/rolldown#1980] [Core Minutes]
In the `rollup` CLI, the camelCase options remain unchanged, which may not be as user-friendly. I propose that we convert the camelCase options to kebab-case in the CLI, for example, changing `externalLiveBindings` to `--external-live-bindings` instead of `--externalLiveBindings`, and also allow us to use `--no-external-live-bindings` to disable the options when are defaultly `true`.

---

- Node.js, Esbuild, Rustc, CSpell, Python, Brew, Grep, **Git**, **npm**, etc. uses `kebab-case`.
- TSC, Vite, Rollup uses `camelCase`.

**Comment by IWANABETHATGUY** [comment]
> In the rollup CLI, the camelCase options remain unchanged, which may not be as user-friendly. 

Can you explain why?

**Reply by 7086cmd** [reply]
1. People often use kebab case when they’re using the command line, so they might just type it in directly.
2. It’s a bit more tricky to type the uppercase letters compared to the hyphen `-`. For the command line, people usually want to type the command as quickly as possible.
3. The “no-“ prefix can also be considered the “kebab-case,” although it may present some challenges. However, we can promptly address these issues.
